src/luaState.cpp

```diff
@@ -144,7 +144,47 @@ void LuaState::setHook(Callable hook, int mask, int count) {
 // Returns true if a lua function exists with the given name
 bool LuaState::luaFunctionExists(String functionName) {
 	// LuaJIT does not return a type here
-	lua_getglobal(L, functionName.ascii().get_data());
+#ifndef LAPI_LUAJIT
+	lua_pushglobaltable(L);
+#else
+	lua_pushvalue(L, LUA_GLOBALSINDEX);
+#endif
+#ifndef LAPI_GDEXTENSION
+	Vector<String> strs = functionName.split(".");
+#else
+	PackedStringArray strs = functionName.split(".");
+#endif
+	for (String str : strs) {
+		if (lua_type(L, -1) != LUA_TTABLE) {
+			lua_pop(L, 1); // Please do not do unforeseen consequences thanks.
```

Comment:
This should be tested for, if there are no issues related to this the comment should be removed.

Reply:
I wrote that before running the test (which i let github actions take care of it) and now since it did pass it will be removed.